src/controllers/oauth.js

```diff
@@ -11,19 +17,35 @@ const oauth = new OAuthServer({ model: authModel })
 const oauthController = module.exports = {}
 
 oauthController.getAuthorizePage = (req, res, next) => {
-  // get query params to find out what the app is (client_id is required)
   if (!req.query || !req.query.client_id) {
-    return res.boom.badRequest('no client_id provided')
+    return res.boom.badRequest(errMessages.NO_CLIENT_ID)
+  }
+  if (!req.query.redirect_uri) {
+    return res.boom.badRequest(errMessages.NO_REDIRECT_URI)
+  }
+  if (!req.query.state) {
+    return res.boom.badRequest(errMessages.NO_STATE)
   }
-  Client.findOne({ id: req.query.client_id })
+
+  if (!req.user) {
+    const queries = {
+      client_id: req.query.client_id,
```

Comment:
What about `redirect_uri` and `state`?

Reply:
@des-des It's all contained in the `return_to` variable, including `client id` - it is just included also as a seprate variable in order to use it to get the name of client as well  (see the other PR about login redirect)